### PR TITLE
fix: update sequence indicators to have visual focus styling

### PR DESCRIPTION
**Pull request by eljefe223**
# Description
closes: #2682 

Added an `after` pseudo element to handle focus visual. Updated stories to use simple state handler to prevent autoplay when carousel is focued.

Before:
Screen Shot 2020-02-20 at 12 18 44 PM

After:
image

Screen Shot 2020-02-20 at 12 18 25 PM

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [x] **Bug fix**: A change that fixes an issue, link to the issue above.
- [ ] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

## Process & policy checklist

- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast-dna/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/en/contributing/standards) for this project.

### CODE REVIEW

File: packages/fast-components-styles-msft/src/carousel/index.ts [hunk 1]
```diff
@@ -80,10 +93,26 @@ const styles: ComponentStyles<CarouselClassNameContract, DesignSystem> = {
...
         display: "inline-block",
-        padding: "0 2px",
+        padding: "4px",
+        position: "relative",
         "&:focus": {
             outline: "none",
         },
+        ...applyFocusVisible<DesignSystem>({
+            "&::after": {
+                opacity: "1",
```
Comment (chrisdholt) [inline packages/fast-components-styles-msft/src/carousel/index.ts]: Is opacity needed here for some other state where it's hidden, or is this the result of copy/paste :)?

File: packages/fast-components-styles-msft/src/carousel/index.ts [hunk 2]
```diff
@@ -80,10 +93,26 @@ const styles: ComponentStyles<CarouselClassNameContract, DesignSystem> = {
...
+                position: "absolute",
+                border: "2px solid transparent",
```
Comment (chrisdholt) [inline packages/fast-components-styles-msft/src/carousel/index.ts]: should this map to the design system? Is the border for the indicators mapped? If so, I think this should follow as well.

File: packages/fast-components-styles-msft/src/carousel/index.ts [hunk 3]
```diff
@@ -80,10 +93,26 @@ const styles: ComponentStyles<CarouselClassNameContract, DesignSystem> = {
...
+                "border-radius": "40px",
+                top: "0",
+                right: "0",
+                left: "0",
+                bottom: "0",
+                content: "''",
+                display: "block",
+                transition: "all 0.05s ease-in-out",
```
Comment (chrisdholt) [inline packages/fast-components-styles-msft/src/carousel/index.ts]: Can we check with design on the transition value here, this seems barely perceivable at 50 milliseconds.

File: packages/fast-components-styles-msft/src/carousel/index.ts [hunk 4]
```diff
@@ -80,10 +93,26 @@ const styles: ComponentStyles<CarouselClassNameContract, DesignSystem> = {
     },
     carousel_sequenceIndicator: {
         display: "inline-block",
-        padding: "0 2px",
+        padding: "4px",
```
Comment (chrisdholt) [inline packages/fast-components-styles-msft/src/carousel/index.ts]: Is this change related or unrelated?

File: packages/fast-components-styles-msft/src/carousel/index.ts [hunk 5]
```diff
@@ -66,7 +79,7 @@ const styles: ComponentStyles<CarouselClassNameContract, DesignSystem> = {
     carousel_slides: {},
     carousel_sequenceIndicators: {
         position: "absolute",
-        bottom: "8px",
+        bottom: "4px",
```
Comment (chrisdholt) [inline packages/fast-components-styles-msft/src/carousel/index.ts]: Is this change related or unrelated?

File: packages/fast-components-styles-msft/src/carousel/index.ts (hunk 2 above)
Comment (eljefe223) [reply packages/fast-components-styles-msft/src/carousel/index.ts]: I wasn't sure entirely. @bheston The sequence indicators have a hard coded width and the height maps to `designUnit` since these elements are so close to them bottom what happens when density increases? Seems that if the height and focus border increase they will clip outside the bounds of the carousel or get very close.

Just bumped the design to 3 and carousel looks very wrong:
image

**Comment by awentzel** [comment]
I would suggest updating the title to express what the fix is as opposed to what the problem is.  For example, fix: update sequence indicators to have visual focus styling or something to that effect.  this way when reading the changelog file, it's clear on exactly what change has been implemented.  Most people don't search for changelog's for problems and this would improve our capability in the future when we surface more dynamic meaningful changelog history.

File: packages/fast-components-styles-msft/src/carousel/index.ts [hunk 6]
```diff
@@ -41,6 +47,13 @@ const lightModeNeutralOutlineRest: DesignSystemResolver<string> = neutralOutline
     (): string => white
 );
 
+const darkModeNneutralFocus: DesignSystemResolver<string> = neutralFocus(
```
Comment (bheston) [inline packages/fast-components-styles-msft/src/carousel/index.ts]: Can we call these `themeDark` and `themeLight` instead to match the style names below. Those names were meant to be based on the setting for the content of the slides rather than the overall UI color, which we have reserved "mode" for.

File: packages/fast-components-styles-msft/src/carousel/index.ts [hunk 7]
```diff
@@ -172,11 +200,17 @@ const styles: ComponentStyles<CarouselClassNameContract, DesignSystem> = {
...
             },
+            "&::after": {
+                "border-color": darkModeNneutralFocus,
+            },
             "&$carousel_sequenceIndicator__active": {
                 "&::before": {
                     background: darkModeNeutralFillStealthRest,
                     ...highContrastHighlightColorIndicator,
                 },
+                "&::after": {
```
Comment (bheston) [inline packages/fast-components-styles-msft/src/carousel/index.ts]: Isn't / shouldn't this be inherited? Any way to not declare twice?
